sprinter/formula/template.py

```diff
@@ -65,9 +70,32 @@ def __install_file(self, config):
                 source_content = lib.cleaned_request('get', source).text
         else:
             source_content = open(os.path.expanduser(source)).read()
+
+        # replace {key} type markers in the template source
+        if config.has('replacement_keys'):
+            replacements = {}
+            for key in config.get('replacement_keys').split(','):
+                key = key.strip()
+                if config.has(key):
+                    replacements[key] = config.get(key)
+            source_content = source_content.format(**replacements)
+
         target_file = os.path.expanduser(config.get('target'))
         parent_directory = os.path.dirname(target_file)
         if not os.path.exists(parent_directory):
             os.makedirs(parent_directory)
+
+        # backup the template, if it is configured for it and if it has changed
+        if os.path.isfile(target_file) and config.has('backup'):
+            target_content = open(target_file).read()
+            if target_content != source_content:
+                backup_target_base = "{path}.{ext}".format(path=target_file, ext=config.get('backup'))
+                backup_target = backup_target_base
+                i = 1
+                while os.path.isfile(backup_target):
+                    backup_target = "{path}-{i}".format(path=backup_target_base, i=i)
+                    i += 1
+                self.logger.info("Backing up template target to %s..." % backup_target)
+                os.rename(target_file, backup_target)
```

Comment:
Possibly overkill, but I'd hate for someone to manually tweak a file and then end up loosing their tweaks when updating.

Thinking about it now, it might be feasible to build the template using the source manifest and only backup if it doesn't match the existing target. That way we don't backup for simple value changes. ...maybe I'll look into that, another time.